libafl_unicorn/src/emu.rs

```diff
@@ -0,0 +1,115 @@
+use capstone::{
+    arch::{self, BuildsCapstone, BuildsCapstoneSyntax},
+    Capstone,
+};
+pub use libafl_targets::{EDGES_MAP, EDGES_MAP_PTR};
+use unicorn_engine::{
+    unicorn_const::{Arch, Permission},
+    RegisterARM, RegisterARM64, RegisterX86, Unicorn,
+};
+
+use crate::helper::get_stack_pointer;
+// TODO: For some reason, the compiled program start by substracting 0x10 to SP
+
+pub fn memory_dump(emu: &Unicorn<()>, len: u64) {
+    let sp = get_stack_pointer(emu);
+    for i in 0..len {
+        let pos = sp + i * 4 - len * 4;
+
+        let data = emu.mem_read_as_vec(pos, 4).unwrap();
+
+        println!(
+            "{:X}:\t {:02X} {:02X} {:02X} {:02X}  {:08b} {:08b} {:08b} {:08b}",
+            pos, data[0], data[1], data[2], data[3], data[0], data[1], data[2], data[3]
+        );
+    }
+}
+
+pub fn debug_print(emu: &Unicorn<()>) {
```

Comment:
Should we provide some helpers or should I move those functions to the example file `fuzzers/full_system/unicorn`?

Reply:
As you prefer. I think it can stay here.

But you want to keep it here, then can you change this to use log::debug! instead?